staging/src/k8s.io/component-base/metrics/registry.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"github.com/blang/semver"
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+	apimachineryversion "k8s.io/apimachinery/pkg/version"
+	"sync"
+)
+
+var globalRegistryFactory = metricsRegistryFactory{
```

Comment:
@brancz is this really the pattern we want to follow again with a new library? 

The alternative: let components create either their own global var or let them parse the registry around.

Reply:
I'd prefer if not, but I don't see a method of migrating that isn't deeply invasive throughout the entire codebase. I 100% agree we should not have globals, but also feel removing this is orthogonal to the goal of this. I do agree that as a next step removing the necessity for this global registry is something we should tackle.

Reply:
I discussed with @logicalhan in Slack: we came up with a package `legacyregistry`, similar to the global scheme we have. That way it is super visible who uses that registry, while it is also clear, that this is probably not the last word.

Reply:
Yes I like that that implies there is change coming.

Reply:
Yeah, it seems like a reasonable compromise to me as well.

Reply:
I also like @sttts' other suggestion of letting components create their own global vars within their own package scope. Either way, I think it's probably good to have a legacy global registry in place as a fallback, since it enables a smoother migration path. 

Anyhow, legacy registry changes are in, please take a look whenever you have a chance. Thanks!